### PR TITLE
Support Jinja in option "mapping"

### PR DESCRIPTION
**Pull request by iSOLveIT**
The feature will allow the mappings option to accept more complex scenarios for the option value. For Instance:

- static value: - e.g. `"spec"`
- dynamic value (coming from excel): - e.g. `"{{status}}"`
- static and dynamic_value: - e.g. `"is_{{status}}"` or `"{{status}}_issue"`
- multiple dynamic values + static: - e.g. `"{{title}}_{{id}}_{{assignee}}"`

Tasks

- [x] Implement Jinja support
- [x] Make values from the excel row available in the jinja context
- [ ] Document this feature and some simple and complex examples

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -196,11 +196,11 @@ or for imports into your Sphinx project.
 
          sne import jira_config --query "project = my_project"
 
-After this you can use ``sne render`` to create e.g. rst-files for your Sphinx project based on own templates.
+After this you can use ``sne render`` to create files, e.g. rst-files, for your Sphinx project based on your templates.
 
 .. code-block:: bash
 
-   sne render -j needs.json -t my_template.rst -o output.rst
+   sne render -j needs.JSON -t my_template.rst -o output.rst
```
Comment (danwos) [inline docs/index.rst:203]: why uppercase JSON?

Looks strange for me in a cmd example

File: docs/services/excel.rst [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+`Sphinx-Needs <https://sphinxcontrib-needs.readthedocs.io/en/latest/>`__.
+
+The ``Excel`` service allows to retrieve external data from spreadsheets during documentation build and
+to create Sphinx-Needs objects based on this data.
+After the creation of the Sphinx-Needs objects, it supports every function from
+`Sphinx-Needs <https://sphinxcontrib-needs.readthedocs.io/en/latest/>`__, which includes Filtering, Linking,
+Updating and much more.
+
+Options
+-------
```
Comment (danwos) [inline docs/services/excel.rst:17]: Maybe describe `content` option as well?

File: docs/snippets/excel_config.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# Manipulates the content to add a link to the source issue
```
Comment (danwos) [inline docs/snippets/excel_config.py:1]: ?? Where gets the link added in this code?

File: sphinx_needs_enterprise/extensions/extension.py [hunk 4]
```diff
@@ -180,10 +180,20 @@ def _extract_data(self, data, options):
...
                     )
                 if isinstance(selector, str):
-                    # Set the "hard-coded" string as value
-                    need_values[name] = selector
+                    # Combine "hard-coded" string and dynamic value and set as value
+                    if re.search(r"{{\w+?}}", selector) is not None and isinstance(
+                        re.search(r"{{\w+?}}", selector), re.Match
+                    ):
+                        need_values[name] = jinja_parse(item, selector)
+                    else:
```
Comment (danwos) [inline sphinx_needs_enterprise/extensions/extension.py:188]: I think  you can always set the value of `jinja_parse()`.

If not ``{{...}`` is in the string/selector, jinja will return the untouched string, which you also do in the else-part.

File: sphinx_needs_enterprise/services/excel.py [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+        mappings_default = {
+            "id": ["id"],
+            "type": "spec",
+            "status": ["status"],
+            "title": ["title"],
+        }
+        dict_undefined_set(config, "mappings", mappings_default)
+
+        # mappings_replaces_default = {
```
Comment (danwos) [inline sphinx_needs_enterprise/services/excel.py:47]: comment can be removed

File: sphinx_needs_enterprise/services/excel.py [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+            if "description" not in datum or datum["description"] is None:
+                datum["description"] = ""
+
+        need_data = self._extract_data(data, options)
+
+        return need_data
+
+    def debug(self, options):
+        # params = self._prepare_request(options)
+        # request_params = {
```
Comment (danwos) [inline sphinx_needs_enterprise/services/excel.py:94]: same here, remove comment

**Review by danwos — CHANGES_REQUESTED** [review]
Great work.
Only some minor stuff.

But I love your coding and documentation style . :+1:

File: sphinx_needs_enterprise/extensions/extension.py (hunk 4 above)
Comment (iSOLveIT) [reply sphinx_needs_enterprise/extensions/extension.py]: To ensure that jinja will return the untouched string if `{{...}}` is not in the string/selector, I made changes to this line:
```python
if isinstance(selector, str):
    # Set the "hard-coded" string or
    # combine the "hard-coded" string and dynamic value
    selector = jinja_parse(item, selector)
    # Set the returned string as value
    need_values[name] = selector

```

**Review by danwos — CHANGES_REQUESTED** [review]
One finding.
The rest is all good. Thanks.

File: sphinx_needs_enterprise/services/excel.py [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+            # Be sure "description" is set and valid
+            if "description" not in datum or datum["description"] is None:
+                datum["description"] = ""
+
+        need_data = self._extract_data(data, options)
+
+        return need_data
+
+    def debug(self, options):
+        file_path = options.get("file", str(self.config["file"]))
```
[the remark this answers is not in the document]
Comment (iSOLveIT) [reply sphinx_needs_enterprise/services/excel.py:85]: > But maybe the "debug" output could be just a table, with the head and the data.
> No Sphinx-Needs objects.

I think that will be better.

**Review by danwos — APPROVED** [review]

**Pull request merged**